### ISSUE TITLE
terminal is misbehaving in the 

### ISSUE THREAD

**Issue by adriatic** [monterey-framework/monterey#161]
![image](https://cloud.githubusercontent.com/assets/2712405/17717833/4287a5d4-63de-11e6-9b88-34a974581a4a.png)

The lines are insterted at the top, scrollbar keeps showing and disappearing result with a very turbulent experience

When finished - I get only this (which is meaningless / useless) as a history of this run

```
[6:23:01 PM] -----FINISHED-----
[6:23:01 PM] [process] finished installing...
[6:23:01 PM] [warn] Not compatible with your operating system or architecture: fsevents@1.0.14
[6:23:01 PM] [warn] Skipping failed optional dependency /chokidar/fsevents:
[6:23:01 PM] [silly] printInstalled
[6:22:45 PM] [warn] npm@3.10.6 should be installed with -g
[6:22:43 PM] [warn] marked@0.3.6 should be installed with -g
[6:20:35 PM] [warn] tough-cookie@2.2.2: ReDoS vulnerability parsing Set-Cookie https://nodesecurity.io/advisories/130
[6:19:43 PM] [process] installing...
[6:19:43 PM] [process] loaded
[6:19:43 PM] [process] chdir: C:\work\tests\Testing monterey\default-esnext
Close
```

**Comment by JeroenVinke** [comment]
If you mean that many messages are not shown after completion then this is by design. NPM outputs many (like thousands) of messages that are not relevant to the user. This was causing monterey to crash from the amount of messages (it could not keep up). In normal consoles I noticed that they did not keep messages of level "silly" or "info" on the screen. Only warnings and errors would be kept on the screen. And this is how it should be, as otherwise you would not be able to see warnings between all the useless messages

If the expectation was to see a tree of installed items, then I think I saw it get printed to the chrome console of monterey once, so I think we can output this on the screen

**Comment by adriatic** [comment]
> If you mean that many messages are not shown after completion then this is by design

No, that is not what I meant. I am just observing that from all thousands of messages I ended up having just 11 (eleven, so you do not think I made a type), none of which amount to anything. **Also** the last message should not be on the top of the terminal, lines should start appearing from the top and get appended after the last one - so the scrolling should "go" from the bottom to the top as it always does.

**Comment by adriatic** [comment]
In particular, I am not sure whether seeing the "tree of installed items" is of any real value - it is almost always to complex to be able to just visually decide some meaning.

My real preference would be to capture everything (pipe it to a file) and then offer a search tool that can be used to find some problems later.

**Comment by JeroenVinke** [comment]
Without newlines (couldn't get that working, but should work if monterey pipes it to a file) the output is 9.4MB:

[log.txt](https://github.com/monterey-framework/monterey/files/422736/log.txt)

you can get the same by running `npm install -ddd`. 

Even though the warnings/errors of npm are good enough for me, i'm OK with adding this feature. But it should be opt-in in order not to waste any more of the users disk space, and so it does not slow down all npm install's

**Comment by adriatic** [comment]
I just downloaded this file (it took 2 seconds) and opened it in WordPad (another second). Then I scrolled up and down without any hesitations (I am always having task manager opened and it did not show any signs of being busy in addition to what is already going).

I wrote to say how the "overhead" of keeping 9 MB files  around for a while to ensure that I can **see** what transpired will give me a lot of comfort. I still really suffer seeing all these npm node-gyp error messages that scroll like crazy and completely hide something that I might really like to see. 

So, this feature allows me postprocessing something of importance

**Comment by adriatic** [comment]
One of the key features of Monterey is **convenience**. Sure that I could do manually **everything** that Monterey does - but it would not be a reliable process

**Comment by JeroenVinke** [comment]
Interesting fact here is that NPM does this already. When an error occurs it will write an `npm-debug.log` file in the target folder with the entire log
